service/src/main/java/tds/session/services/impl/SessionServiceImpl.java

```diff
@@ -23,4 +23,11 @@ public SessionServiceImpl(SessionRepository sessionRepository) {
     public Optional<Session> getSessionById(UUID id) {
         return sessionRepository.getSessionById(id);
     }
+
+    @Override
+    public void pause(final UUID sessionId, final String newStatus) {
+        // TODO:  Throw an exception if attempting to pause a Session that is already paused?
```

Comment:
Small request, in the future can we move the off topic comment to the overall comment section as not to mix threads?

That being said this is mixed so I think if exams need to be paused when a session is paused then I would expect session to work in the following order:

1. Check if session is already paused.  If so return.
2. Pause exams calling exam microservice
3. If 200 from previous step pause session
4. return success for pausing the session